### ISSUE TITLE
Inference question about sinkhorn algorithm.

### ISSUE THREAD

**Issue by wqdong8** [zbr17/OptVQ#8]
Hi, thanks for sharing this great work!

I have a quick question: after OptVQ training, is it necessary to run the Sinkhorn algorithm during inference? 
Or can we simply use something like:
```
def quantize_input(self, query, reference):
    # compute the distance matrix
    query2ref = torch.cdist(query, reference, p=2.0) # (B1, B2)

    # find the nearest embedding
    indices = torch.argmin(query2ref, dim=-1) # (B1,)
    nearest_ref = reference[indices] # (B1, C)
        
    return indices, nearest_ref, query2ref
```
as in naive VQ-VAE?

Thanks in advance!

**Comment by zbr17** [comment]
Hello! The Sinkhorn algorithm shares some similar properties with the Batch Normalization (Batch-norm) layer. That is to say, when the number of data points varies during the inference process, the results of the Sinkhorn algorithm are likely to exhibit subtle differences. Currently, we have conducted some research on this issue. Additionally, in another issue (https://github.com/zbr17/OptVQ/issues/3), we have provided some visualization results. We have found that when either the Sinkhorn algorithm or the nearest neighbor method is used during inference, the differences in the reconstructed results are not significant. **Therefore, you can use the `nearest search` method during inference.**

**Comment by wqdong8** [comment]
Understood, thank you again for the explanation!
